service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java

```diff
@@ -359,20 +358,15 @@ public void testGetRegistrationAssertionSucceedsForExistingDevice(final VertxTes
         when(registrationClient.assertRegistration(eq("device"), any())).thenReturn(Future.succeededFuture(assertionResult));
         when(registrationClient.assertRegistration(eq("device"), any(), any())).thenReturn(Future.succeededFuture(assertionResult));
 
-        final Checkpoint assertion = ctx.checkpoint();
         // WHEN an assertion for the device is retrieved
         adapter.getRegistrationAssertion("tenant", "device", null, mock(SpanContext.class))
-                .onComplete(ctx.succeeding(result -> ctx.verify(() -> {
-            // THEN the result contains the registration assertion
-            assertEquals(assertionResult, result);
-            assertion.flag();
-        })));
-        adapter.getRegistrationAssertion("tenant", "device", null, mock(SpanContext.class))
-                .onComplete(ctx.succeeding(result -> ctx.verify(() -> {
-            // THEN the result contains the registration assertion
-            assertEquals(assertionResult, result);
-            assertion.flag();
-        })));
+                .onComplete(ctx.succeeding(result -> {
```

Comment:
we should also remove the duplicate recording on the registrationClient for the variant that is no longer invoked by AbstractProtocolAdapterBase ...